### PR TITLE
Add description metadata to Grafana Agent docs (#5296)

### PR DESCRIPTION
**Pull request by clayton-cornell**
* Add description metadata

* Add description metadata to shared content

* Fix the title metadata in fmt

* Update the doc generator tool

* Clean up some metadata errors

(cherry picked from commit d529b37a70a1f6055a7dde803043446c3e0ffd38)

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/operator/release-notes.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ---
 canonical: https://grafana.com/docs/agent/latest/operator/release-notes/
 description: Release notes for Grafana Agent static mode Kubernetes operator
-title: Release notes for Grafana Agent
 menuTitle: Release notes
-aliases:
-- ./upgrade-guide/
```
Comment (jdbaldry) [inline docs/sources/operator/release-notes.md:7]: Is the removal of this alias intended?

File: docs/sources/static/configuration/scraping-service.md [hunk 2]
```diff
@@ -149,8 +151,8 @@ metrics:
               - etcd:2379
 ```
 
-Note that there are no instance configs present in this example; instance
-configs must be passed to the API for the Agent to start scraping metrics.
+Note that there are no instance configurations present in this example; instance
+configurationss must be passed to the API for the Agent to start scraping metrics.
```
Comment (jdbaldry) [inline docs/sources/static/configuration/scraping-service.md:155]:
```suggestion
configurations must be passed to the API for the Agent to start scraping metrics.
```

**Review by jdbaldry — APPROVED** [review]
Spotted a typo and have one question but approving because the descriptions look good at a scan.

File: docs/sources/operator/release-notes.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/operator/release-notes.md:7]: Ooops, no that was a merge conflict resolution thing I missed :-( I'll fix it.

File: docs/sources/operator/release-notes.md [hunk 3]
```diff
@@ -1,14 +1,14 @@
 ---
 canonical: https://grafana.com/docs/agent/latest/operator/release-notes/
 description: Release notes for Grafana Agent static mode Kubernetes operator
-title: Release notes for Grafana Agent
 menuTitle: Release notes
```
Comment (clayton-cornell) [inline docs/sources/operator/release-notes.md:4]:
```suggestion
menuTitle: Release notes
aliases:
- ./upgrade-guide/
```

**Pull request merged**